forward_command_controller/src/multi_interface_forward_controller.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2021, PickNik, Inc.
+// Copyright (c) 2021, Stogl Robotics Consulting UG (haftungsbeschränkt) (template)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "forward_command_controller/multi_interface_forward_controller.hpp"
+
+#include <string>
+#include <vector>
+
+namespace forward_command_controller
+{
+
+MultiInterfaceForwardController::MultiInterfaceForwardController()
+: forward_command_controller::ForwardCommandController()
+{
+  logger_name_ = "multi interface forward controller";
+}
+
+controller_interface::return_type
+MultiInterfaceForwardController::init(const std::string & controller_name)
+{
+  auto ret = ForwardCommandController::init(controller_name);
+  if (ret != controller_interface::return_type::SUCCESS) {
+    return ret;
+  }
+
+  try {
+    node_->undeclare_parameter("joints");
```

Comment:
Seeing this I'm tempted to say we shouldn't even inherit from ForwardCommandController. It is not a specialization of a controller if you rip all parameters out and replace them with new ones.

Reply:
I partially agree, but there is a big overlap on the other side. What do you think to add a virtual ForwardController without parameters and then inheriting from it for both, `ForwardCommandController` and `MultiInterfaceForwardController`?